### DISCUSSION TITLE
How to limit data collected Do While task?

### DISCUSSION THREAD

**Discussion by kyle-c9000** [Netflix/conductor#2557] [Q&A]
I am using a Do While task and it gradually collects more and more data into the workflow object every iteration until it gets so large it becomes a problem. Since I only really need the data from the latest iteration I am wondering if there is a good way to only keep the last iteration or last X iterations.

**Comment by aravindanr** [comment]
This is not currently supported, however, it can be supported by adding an additional property in the task that specifies how many iterations need to be kept. With our current roadmap, we won't be able to get this soon. If you are inclined, you can create a pull request to support this feature,

**Comment by kyle-c9000** [comment]
Ok, this might be something I can add, do you have any suggestions on what Classes/functions would need to be modified besides the change to the task definition?
